src/components/Explore/BalanceSummary.tsx

```diff
@@ -63,32 +70,34 @@ export default function BalanceSummary({ address }: { address: string }) {
             </TotalBalance>
           </TotalBalanceSection>
           <NetworkBalancesSection>Your balances by network</NetworkBalancesSection>
-          <NetworkBalance
-            logoUrl={logoUrl}
-            balance={'1'}
-            tokenSymbol={tokenSymbol ?? 'XXX'}
-            fiatValue={fiatValue}
-            label={label}
-            networkColor={theme.primary1}
-          />
-          <NetworkBalance
-            logoUrl={logoUrl2}
-            balance={'3.3'}
-            tokenSymbol={tokenSymbol ?? 'XXX'}
-            fiatValue={3200}
-            label={'Polygon'}
-            networkColor={TEXT_COLORS['137']}
-          />
+          {data &&
+            Object.entries(data).map(([chainId, amount]) => {
+              const fiatValue = 1010.12 // for testing purposes
+              const chain = parseInt(chainId)
+              if (!fiatValue || !isChainAllowed(connector, chain)) return null
```

Comment:
why do we need to check if the chain is allowed if there's data already?

Reply:
defensive safety check since the UI doesn't control the data source. 

i.e., if the data service started returning Celo balances before the UI had a `chainInfo` for it things could explode